ai_server/server.py

```diff
@@ -1,12 +1,250 @@
+LLAMA_CPP_MODEL_DIRS = [
+    'DeepSeek-V2.5-IQ1_M',
+    'DeepSeek-V3-0324-UD-IQ2_XXS', 
+    'DeepSeek-V3-0324-UD-Q2_K_XL'
+]
+
+# Output parsing skip patterns for llama-cli
+LLAMA_CPP_SKIP_PATTERNS = [
+    'ggml_cuda_init:', 'warning:', 'build:', 'main:', 'llama_model_loader:',
+    'print_info:', 'load:', 'load_tensors:', 'llama_context:', 
+    'common_init_from_params:', 'system_info:', 'sampler', 'generate:'
+]
+
+
+def chat_with_llama_server_http(model: str, content: str, timeout: int = 300) -> str:
+    """Handle chat using llama-server HTTP API."""
+    if not LLAMA_SERVER_URL:
+        raise Exception("LLAMA_SERVER_URL environment variable not set")
+    
+    try:
+        response = requests.post(
+            f'{LLAMA_SERVER_URL}/v1/chat/completions',
+            json={
+                'model': model,
+                'messages': [{'role': 'user', 'content': content}],
+                'stream': False,
+                'max_tokens': 512
+            },
+            headers={'Content-Type': 'application/json'},
+            timeout=timeout
+        )
+        
+        if response.status_code == 200:
+            data = response.json()
+            if 'choices' in data and len(data['choices']) > 0:
+                return data['choices'][0]['message']['content']
+            else:
+                raise Exception("Invalid response format from llama-server")
+        else:
+            response = requests.post(
+                f'{LLAMA_SERVER_URL}/completion',
+                json={
+                    'prompt': content,
+                    'n_predict': 512,
+                    'stream': False
+                },
+                headers={'Content-Type': 'application/json'},
+                timeout=timeout
+            )
+            
+            if response.status_code == 200:
+                data = response.json()
+                return data.get('content', 'No response generated.')
+            else:
+                raise Exception(f"Llama-server HTTP error: {response.status_code} - {response.text}")
+                
+    except requests.Timeout:
+        raise Exception(f"Llama-server request timed out for model {model}")
+    except requests.RequestException as e:
+        raise Exception(f"Llama-server request failed: {str(e)}")
+
+def get_available_llamacpp_models():
+    """Return list of available llama.cpp models."""
+    models = []
+    for model_dir in LLAMA_CPP_MODEL_DIRS:
+        dir_path = os.path.join(GGUF_DIR, model_dir)
+        if os.path.exists(dir_path):
+            try:
+                files = os.listdir(dir_path)
+                for file in files:
+                    if file.endswith('.gguf'):
+                        models.append(f"{model_dir}/{file}")
+            except Exception:
+                continue
+    return models
+
+def resolve_model_path(model: str) -> Optional[str]:
+    """Resolve model name to full GGUF file path."""
+    for model_dir in LLAMA_CPP_MODEL_DIRS:
```

Comment:
Overall I think this idea is good but this implementation is trying to do a bit too much.

Given an environment variable `LLAMA_MODL_DIR=/data1/GGUF`, you can construct a *glob expression* of the form `/data1/GGUF/<model name>/*.gguf` and return the first file that matches this expression, or `None` if there isn't such a path.